sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -0,0 +1,298 @@
+    {
+        { "key" ,"value" }
+    },
+    IpConfigurations = new List<NetworkInterfaceIPConfiguration>()
+    {
+        new NetworkInterfaceIPConfiguration()
+        {
+            Name = ipConfigName,
+            PrivateIPAllocationMethod = IPAllocationMethod.Dynamic,
+            Subnet = subnetResponse,
+        }
+    },
+    NetworkSecurityGroup = nsg
+};
+
+NetworkInterface putNicResponse = networkClient.NetworkInterfaces.CreateOrUpdate(rgName, nicname, nicParameters);
+NetworkInterface nicResponse = networkClient.NetworkInterfaces.Get(rgName, nicname);
+```
+#### New
+```C#
+string nicname = vmName + "_nic";
+NetworkInterfaceData nicData = new NetworkInterfaceData()
+{
+    Location = location,
+    IpConfigurations = new List<NetworkInterfaceIPConfiguration>()
+    {
+        new NetworkInterfaceIPConfiguration()
+        {
+            Name = "Primary",
+            Primary = true,
+            Subnet = new Subnet() { Id = vnet.Subnets.First().Id },
+            PrivateIPAllocationMethod = IPAllocationMethod.Dynamic,
+            PublicIPAddress = new PublicIPAddress() { Id = ipAddress.Id }
+        }
+    }
+};
+NetworkInterface nic = (await networkInterfaceContainer.CreateOrUpdateAsync(nicName, nicData)).Value;
+```
+
+This step is similar to the old SDK, however, notice that the `CreateOrUpdateAsync()` method returns the network interface that has been created. 
+
+### Create a Virtual Machine
+#### Old
+```C#
+string vmSize = VirtualMachineSizeTypes.StandardA1V2
+VirtualMachine inputVM = new VirtualMachine
+{
+    Location = location,
+    Tags = new Dictionary<string, string>() { { "RG", "rg" }, { "testTag", "1" } },
+    HardwareProfile = new HardwareProfile
+    {
+        VmSize = vmSize
+    },
+    StorageProfile = new StorageProfile
+    {
+        ImageReference = imageRef,
+        OsDisk = new OSDisk
+        {
+            Caching = CachingTypes.None,
+            WriteAcceleratorEnabled = writeAcceleratorEnabled,
+            CreateOption = DiskCreateOptionTypes.FromImage,
+            Name = "test",
+            Vhd =  null,
+            ManagedDisk = new ManagedDiskParameters
+            {
+                StorageAccountType = osDiskStorageAccountType,
+                DiskEncryptionSet = null,
+            }
+        },
+        DataDisks = null,
+    },
+    OsProfile = new OSProfile
+    {
+        AdminUsername = "Foo12",
+        AdminPassword = PLACEHOLDER,
+        ComputerName = ComputerName
+    }
+};
+
+inputVM.AvailabilitySet = new Microsoft.Azure.Management.Compute.Models.SubResource() { Id = aSetID };
+
+CM.NetworkProfile vmNetworkProfile = new CM.NetworkProfile();
+vmNetworkProfile.NetworkInterfaces = new List<NetworkInterfaceReference>
+{
+    new NetworkInterfaceReference
+    {
+        Id = nicResponse.Id
+    }
+};
+inputVM.NetworkProfile = vmNetworkProfile;
+
+VirtualMachine vm = VMcomputeClient.VirtualMachines.CreateOrUpdate(rgName, inputVM.Name, inputVM);
+```
+#### New
```

Comment:
this doc is great! besides guides on specific resource operation, how about add general operations migration guide, like paging, error handling, customized policy etc? similar like Java migration guide: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/docs/MIGRATION_GUIDE.md